### PR TITLE
Implemented a simplified optimized calculator for the event loss table

### PR DESCRIPTION
**Pull request by micheles**
See https://bugs.launchpad.net/oq-engine/+bug/1465210.
The tests are green: https://ci.openquake.org/job/zdevel_oq-risklib/736

### CODE REVIEW

File: openquake/commonlib/calculators/ebr.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    :param riskinputs:
+        a list of :class:`openquake.risklib.riskinput.RiskInput` objects
+    :param riskmodel:
+        a :class:`openquake.risklib.riskinput.RiskModel` instance
+    :param rlzs_assoc:
+        a class:`openquake.commonlib.source.RlzsAssoc` instance
+    :param monitor:
+        :class:`openquake.commonlib.parallel.PerformanceMonitor` instance
+    :returns:
+        a dictionary rlz.ordinal -> (loss_type, tag) -> AccumDict()
```
Comment (ptormene) [inline openquake/commonlib/calculators/ebr.py]: Please fix the return value

File: openquake/commonlib/calculators/ebr.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            losses[idx] = [numpy.array(dic.items(), elt_dt)]
+        else:
+            losses[idx] = []
+    return losses
+
+
+@base.calculators.add('ebr')
+class EventBasedRiskCalculator(base.RiskCalculator):
+    """
+    Event based PSHA calculator generating the ruptures only
```
Comment (ptormene) [inline openquake/commonlib/calculators/ebr.py]: Please fix the docstring

File: openquake/commonlib/calculators/ebr.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    event_loss_table = datastore.persistent_attribute(
+        '/event_loss_table-rlzs')
+    insured_loss_table = datastore.persistent_attribute(
+        '/insured_loss_table-rlzs')
+    is_stochastic = True
+
+    def pre_execute(self):
+        """
+        Read the precomputed ruptures (or compute them on the fly) and
+        prepare some empty files in the export directory to store the gmfs
```
Comment (ptormene) [inline openquake/commonlib/calculators/ebr.py]: Please fix the docstring

File: openquake/commonlib/calculators/ebr.py [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+                losses = numpy.concatenate(arrays)
+
+                # appending rows to the event loss table
+                n1 = n + len(losses)
+                elt.resize((n1,))
+                saved_mb += losses.nbytes
+                elt[n:n1] = losses
+                pos[i, l, r] = n1  # update position in the elt
+            self.datastore.hdf5.flush()
+            if saved_mb > 1:
```
Comment (ptormene) [inline openquake/commonlib/calculators/ebr.py]: Please log anyway

File: openquake/commonlib/calculators/ebr.py [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+                    continue
+                losses = numpy.concatenate(arrays)
+
+                # appending rows to the event loss table
+                n1 = n + len(losses)
+                elt.resize((n1,))
+                saved_mb += losses.nbytes
+                elt[n:n1] = losses
+                pos[i, l, r] = n1  # update position in the elt
+            self.datastore.hdf5.flush()
```
Comment (ptormene) [inline openquake/commonlib/calculators/ebr.py]: We might flush inside the loop

File: openquake/commonlib/calculators/event_based.py [hunk 6]
```diff
@@ -338,11 +342,14 @@ def post_execute(self, result):
...
         nc = self.rlzs_assoc.csm_info.num_collections
         sescollection = [{} for col_id in range(nc)]
+        tags = []
         for trt_id in result:
             for sr in result[trt_id]:
                 sescollection[sr.col_id][sr.tag] = sr
+                tags.append(sr.tag)
         logging.info('Saving the SES collection')
         with self.monitor('saving ses', autoflush=True):
+            self.tags = numpy.array(sorted(tags), (str, 100))
```
Comment (ptormene) [inline openquake/commonlib/calculators/event_based.py]: Please add a comment about the possibility that the string will be truncated if longer than 100

File: openquake/commonlib/export/risk.py [hunk 7]
```diff
@@ -215,6 +215,27 @@ def export_event_loss(ekey, dstore):
...
+    """
+    name, fmt = ekey
+    fnames = []
+    elt = dstore[name]
+    tags = dstore['/tags']
+    for loss_type in elt:
+        for rlz_uid in elt[loss_type]:
+            data = [[tags[e['rup_id']], e['loss']]
+                    for e in elt[loss_type][rlz_uid]]
+            fname = '%s-%s-%s.csv' % (name[1:-5], rlz_uid, loss_type)
```
Comment (ptormene) [inline openquake/commonlib/export/risk.py]: Please add a comment about how you are building the name

**Comment by ptormene** [comment]
Besides the comments, LGTM

File: openquake/commonlib/calculators/ebr.py (hunk 1 above)
Comment (micheles) [inline openquake/commonlib/calculators/ebr.py]: Done

File: openquake/commonlib/calculators/ebr.py (hunk 2 above)
Comment (micheles) [inline openquake/commonlib/calculators/ebr.py]: Done

File: openquake/commonlib/calculators/ebr.py (hunk 3 above)
Comment (micheles) [inline openquake/commonlib/calculators/ebr.py]: Done

File: openquake/commonlib/calculators/ebr.py (hunk 4 above)
Comment (micheles) [inline openquake/commonlib/calculators/ebr.py]: Done

File: openquake/commonlib/calculators/ebr.py (hunk 5 above)
Comment (micheles) [inline openquake/commonlib/calculators/ebr.py]: Done

File: openquake/commonlib/calculators/event_based.py (hunk 6 above)
Comment (micheles) [inline openquake/commonlib/calculators/event_based.py]: Added a warning

File: openquake/commonlib/export/risk.py (hunk 7 above)
Comment (micheles) [inline openquake/commonlib/export/risk.py]: Done

**Pull request merged**